src/unitxt/evaluate_cli.py

```diff
@@ -0,0 +1,939 @@
+logger = get_logger()
+
+
+def _parse_key_value_string(value: str) -> Optional[Dict[str, Any]]:
+    """Parses a key=value comma-separated string into a dict."""
+    parsed_dict = {}
+    for item in value.split(","):
+        item = item.strip()
+        if not item:
+            continue
+        parts = item.split("=", 1)
+        if len(parts) == 2:
+            key, val_str = parts
+            key = key.strip()  # Strip whitespace from key
+            # Attempt to convert value to int, float, or bool
+            val: Any
+            try:
+                val = int(val_str)
+            except ValueError:
+                try:
+                    val = float(val_str)
+                except ValueError:
+                    if val_str.lower() == "true":
+                        val = True
+                    elif val_str.lower() == "false":
+                        val = False
+                    else:
+                        val = val_str  # Keep as string
+            parsed_dict[key] = val
+        else:
+            logger.warning(
+                f"Could not parse argument part: '{item}'. Expected format 'key=value'."
+            )
+    return parsed_dict if parsed_dict else None
+
+
+def try_parse_json(value: str) -> Union[str, dict, None]:
+    """Attempts to parse a string as JSON or key=value pairs.
+
+    Returns the original string if parsing fails
+    and the string doesn't look like JSON/kv pairs.
+    Raises ArgumentTypeError if it looks like JSON but is invalid.
+    """
+    if value is None:
+        return None
+    try:
+        # Handle simple key-value pairs like "key=value,key2=value2"
+        if "=" in value and "{" not in value:
+            parsed_dict = _parse_key_value_string(value)
+            if parsed_dict:
+                return parsed_dict
+
+        # Attempt standard JSON parsing
+        return json.loads(value)
+
+    except json.JSONDecodeError as e:
+        if value.strip().startswith("{") or value.strip().startswith("["):
+            raise argparse.ArgumentTypeError(
+                f"Invalid JSON: '{value}'. Hint: Use double quotes for JSON strings and check syntax."
+            ) from e
+        return value  # Return as string if not JSON-like
+    except Exception as e:
+        logger.error(f"Error parsing argument '{value}': {e}")
+        # Fixed B904: Added 'from e'
+        raise argparse.ArgumentTypeError(f"Could not parse argument: '{value}'") from e
+
+
+def setup_parser() -> argparse.ArgumentParser:
+    """Sets up the argument parser."""
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description="CLI utility for running evaluations with unitxt.",
+    )
+
+    # --- Task/Dataset Arguments ---
+    parser.add_argument(
+        "--task",
+        "-t",
+        type=str,
+        required=True,
+        help="Unitxt task/dataset identifier string.\n"
```

Comment:
Maybe worth adding link to documentation of the api https://www.unitxt.ai/en/latest/unitxt.standard.html#module-unitxt.standard